web/client/plugins/Details.jsx

```diff
@@ -6,48 +6,118 @@
+    contentChangedSelector,
+    editingSelector,
+    settingsSelector,
+    loadingSelector,
+    loadFlagsSelector,
+    editedSettingsSelector
+} from '../selectors/details';
+import { setControlProperty } from '../actions/controls';
+import { mapFromIdSelector } from '../selectors/maps';
+import { mapInfoSelector, mapIdSelector, mapInfoDetailsUriFromIdSelector } from '../selectors/map';
+import { mapLayoutValuesSelector } from '../selectors/maplayout';
+import { isLoggedIn } from '../selectors/security';
+
+import Message from '../components/I18N/Message';
+import Details from '../components/details/Details';
+import { createPlugin } from '../utils/PluginsUtils';
+
+import details from '../reducers/details';
+import * as epics from '../epics/details';
 
 /**
  * Details plugin used for fetching details of the map
  * @class
  * @memberof plugins
  */
 
-module.exports = {
-    DetailsPlugin: connect((state) => ({
-        active: get(state, "controls.details.enabled"),
+export default createPlugin('Details', {
+    component: connect((state) => ({
+        show: get(state, "controls.details.enabled"),
+        loading: loadingSelector(state),
+        loadFlags: loadFlagsSelector(state),
+        mapInfo: mapInfoSelector(state),
+        canEdit: isLoggedIn(state) && get(mapInfoSelector(state), 'canEdit'),
+        settings: settingsSelector(state),
         map: mapFromIdSelector(state, mapIdSelector(state)),
-        dockStyle: mapLayoutValuesSelector(state, {height: true}),
-        detailsText: currentMapDetailsTextSelector(state)
+        content: contentSelector(state),
+        editedContent: editedContentSelector(state),
+        contentChanged: contentChangedSelector(state),
+        editedSettings: editedSettingsSelector(state),
+        editing: editingSelector(state),
+        dockStyle: mapLayoutValuesSelector(state, {height: true})
     }), {
-        onClose: closeDetailsPanel
-    })(assign(require('../components/details/DetailsPanel'), {
+        onSave: save,
+        onEdit: edit.bind(null, 'content'),
+        onEditSettings: edit.bind(null, 'settings'),
+        onCancelEdit: cancelEdit,
+        onClose: setControlProperty.bind(null, 'details', 'enabled', false),
+        onSettingsChange: changeSetting,
+        onEditorUpdate: setEditedContent
+    }, (stateProps, dispatchProps, ownProps) => ({
+        ...stateProps,
+        ...ownProps,
+        ...dispatchProps,
+        editorProps: {
+            ...(stateProps.editorProps || {}),
+            ...(ownProps.editorProps || {}),
+            onUpdate: dispatchProps.onEditorUpdate
+        }
+    }))(Details),
+    containers: {
         BurgerMenu: {
             name: 'details',
             position: 1000,
             text: <Message msgId="details.title"/>,
             icon: <Glyphicon glyph="sheet"/>,
-            action: openDetailsPanel,
+            action: setControlProperty.bind(null, 'details', 'enabled', true),
+            selector: (state) => {
```

Comment:
here i have some doubts.
if someone removes the Toolbar from localConfig you are not able to reopen the details if 
showAsModal is turned on && showOnStartUp is turned off

In my opinion this plugin should be shown without looking at showAsModal flag.
Another things is that if you add a priority of 1 here and on let's say 5 on the Toolbar container this will make the details plugin visible in the toolbar if both are present, or visible in burger menu if toolbar is not present. (but we still just check the presence of detailsUri

@tdipisa what do you think?